ingestion/functions/parsing/argentina/argentina.py

```diff
@@ -0,0 +1,233 @@
+
+
+def convert_age(entry):
+    '''
+    Want to return a float specifying age in years. If age field is empty, return None
+    '''
+    if entry['edad']:
+        if entry['edad_años_meses'] == 'Años':
+            return float(entry['edad'])
+        elif entry['edad_años_meses'] == 'Meses':
+            return float(float(entry['edad']) / 12)
+    else:
+        return None
+
+
+def get_confirmed_event(entry):
+    if entry['fecha_diagnostico']:
+        confirmation_date = convert_date(entry['fecha_diagnostico'])
+        note = 'Using Date of Diagnosis as the date of confirmation.'
+
+    elif entry['fecha_inicio_sintomas']:
+        confirmation_date = convert_date(entry['fecha_inicio_sintomas'])
+        note = 'Using Date of Symptom Onset as the date of confirmation, because Date of Diagnosis is missing.'
+
+    elif entry['fecha_apertura']:
+        confirmation_date = convert_date(entry['fecha_apertura'])
+        note = 'Using Date of Case Opening as the date of confirmation, because both Date of Diagnosis and Date of Symptom Onset are missing.'
+
+    if 'Caso confirmado por laboratorio' in entry['clasificacion']:
+        confirmed_value = 'Laboratory Test'
+    elif 'Caso confirmado por criterio clínico-epidemiológico' in entry['clasificacion']:
+        confirmed_value = 'Clinical Diagnosis'
+    else:
+        confirmed_value = 'Method Unknown'
+    return confirmation_date, confirmed_value, note
+
+
+def convert_residential_location(entry):
+    '''
+    This gets the residential address of the patient. Note this is not used to locate the case.
+    '''
+    query_terms = [term for term in [
+        entry.get("residencia_provincia_nombre", ""),
+        entry.get("residencia_departamento_nombre", ""),
+        entry.get("residencia_pais_nombre", "")]
+        if term != '']
+
+    return ", ".join(query_terms)
+
+
+def parse_cases(raw_data_file, source_id, source_url):
+    """
+    Parses G.h-format case data from raw API data.
+
+    We are currently only incorporating cases classified ('clasificacion_resumen') as 'Confirmed'. However,
+    970k out of 1.5M cases are listed as 'Discarded', even though many have data values resembling confirmed
+    Covid-19 patients, eg date_of_diagnosis, ICU_admission, mechanical breathing assistance. Future versions may
+    want to modify this behaviour.
+
+    For cases classified as Confirmed but lacking a Date of Diagnosis, we use Date of Symptom onset where present,
+    and Date of Case Opening where neither Date of Diagnosis or Date of Symptom Onset are present.
+
+    For case location, we use 'Province of case loading' (carga_provincia_nombre). This is where
+    the laboratory tests were carried out, so may not always correspond to the exact location of the case, but is
+    best proxy we have. The other location date refers to the residential address of the patient.
+
+    """
+
+    private_public_map = {'Público': 'Public', 'Privado': 'Private'}
```

Comment:
declare this at the top level please (it doesn't require anything from the parsing class)